### PR TITLE
ci: update Go version to go1.16

### PR DESCRIPTION
**Pull request by efd6**

### CODE REVIEW

**Comment by efd6** [comment]
This failure is due to the change in go tool behaviour for installing executables; prior to go1.16, packages and executables are installed with `go get`, while after, executables are installed with the versioned `go install` invocation. The [relevant part](https://github.com/elastic/apm-pipeline-library/blob/c6fa3016fd7220654f8550b17837f41fb7511aac/vars/withGoEnvUnix.groovy#L67-L78) of the APM pipeline library still uses the `go get` invocation, which contaminates the go.mod and go.sum files.

I have confirmed this locally. I'm not sure how to move forward with this since this because a very wide-spread change as `withGoEnv` now needs to be taught about the difference between packages and executables.

**Comment by v1v** [comment]
Can you edit https://github.com/elastic/stream/pull/27/files#diff-ef8b2f5e4d21d81c0eba4754dea3a3dfc2c67d26284f195165b650cf4aff7beaL3 and replace `current` with `master` to double check everything works as expected after your changes in https://github.com/elastic/apm-pipeline-library/pull/1300

Then you can revert it and we can cut a release for https://github.com/elastic/apm-pipeline-library

**Comment by efd6** [comment]
I hope [this](https://github.com/elastic/stream/pull/27/commits/a0fbfc408566b913c54b06db3402dad2d775329c) is what you meant.

**Comment by v1v** [comment]
I'll cut a release now and then we can revert https://github.com/elastic/stream/pull/27/commits/a0fbfc408566b913c54b06db3402dad2d775329c

Thanks for verifying those changes 💯

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env groovy
 
-@Library('apm@current') _
+@Library('apm@master') _
```
Comment (v1v) [inline .ci/Jenkinsfile:3]:
```suggestion
@Library('apm@current') _
```

**Review by v1v — APPROVED** [review]
Let's wait for the new build in the CI and if passes then :100 to merge this.
Thanks for the fix and these changes!!

**Pull request merged**